src/internal/connector/onedrive/collections_test.go

```diff
@@ -33,20 +33,38 @@ type statePath struct {
 func getExpectedStatePathGenerator(
 	t *testing.T,
 	tenant, user, base string,
-) func(data.CollectionState, string) statePath {
-	return func(state data.CollectionState, pth string) statePath {
-		p, err := GetCanonicalPath(base+pth, tenant, user, OneDriveSource)
-		require.NoError(t, err)
-
+) func(data.CollectionState, ...string) statePath {
+	return func(state data.CollectionState, pths ...string) statePath {
 		var (
-			cp path.Path
-			pp path.Path
+			p1  path.Path
+			p2  path.Path
+			pp  path.Path
+			cp  path.Path
+			err error
 		)
 
-		if state == data.NewState {
-			cp = p
+		if state != data.MovedState {
+			require.Equal(t, len(pths), 1, "invalid number of paths to getExpectedStatePathGenerator")
```

Comment:
there's lots of testify functions one can use
```suggestion
			require.Len(t, pths, 1, "invalid number of paths to getExpectedStatePathGenerator")
```